src/builder.ts

```diff
@@ -100,10 +100,14 @@ export default class Builder {
 
 			// Catch any errors the stream produces
 			outputStream.on('error', (err: Error) => {
-				self.callHook(hooks, 'buildFailure', handler, err)
+				let layers = self.layers
+				layers.pop()
```

Comment:
Worth mentioning that `pop()` modifies the original array and I see there are several places where `pop()` is called on the same object and it's then passed around. Are you sure it should be dropping layers on all those cases? Just double checking.